TFUtil.py

```diff
@@ -906,10 +921,11 @@ def get_placeholder_with_specific_batch_dim_axis(self, batch_dim_axis):
   def get_placeholder_time_flattened(self):
     assert self.placeholder is not None
     assert self.have_time_axis()
-    # flatten_with_seq_len_mask only works for these two cases at the moment:
-    assert (self.time_dim_axis, self.batch_dim_axis) == (0, 1) or (self.time_dim_axis, self.batch_dim_axis) == (1, 0)
+    # flatten_with_seq_len_mask only works for these cases at the moment:
+    assert (self.time_dim_axis, self.batch_dim_axis) in [(0, 1), (1, 0), (0, 2), (2, 0)]
```

Comment:
This is too specific. If you want to support any axes for batch/time, why are you checking that it must be 0,2 or 2,0?

Reply:
Actually, it supports other cases, but they have to be checked. The only requirement is that either batch_dim_axis or time_dim_axis is 0.

Reply:
Then check only that.